### ISSUE TITLE
Unification of equations for 2D and 3D (non)linear elasticity

### ISSUE THREAD

**Issue by danielwolff1** [imcs-compsim/pinns_for_comp_mech#23]
At the moment, there is a lot of code duplication when it comes to the implementation of the equations in 2D and 3D (as well as the linear and the nonlinear case). 

For better readability and maintainability of the code, we should aim to remove code duplications and unify the different implementations. 

This however requires a more modular (i.e., dimension-agnostic) approach of handling the actual conservation laws: At the moment, matrix-vector products are hard-coded instead of relying on the chosen backend (i.e., `tensorflow` or `Pytorch`) to do the math. While this is not as problematic for small-deformation elasticity (i.e., the "linear" case), it becomes increasingly important in the nonlinear case, where the equations involve Jacobian determinants and matrix inversions.

A first step into this direction has also already been undertaken by @kappenotto and some functions have been implemented in (`elasticity_utils_3d_form.py`)[https://github.com/imcs-compsim/pinns_for_comp_mech/blob/BA-Pinns_for_nonlinear_problems/utils/elasticity/elasticity_utils_3d_form.py] on branch `BA-Pinns_for_nonlinear_problems`, therefore this issue is related to #22.

**Comment by danielwolff1** [comment]
I just started a new branch for this, where I try to tackle this problem based on the first steps by @kappenotto:

Check out branch [`refactor-elasticity-utils`](https://github.com/imcs-compsim/pinns_for_comp_mech/tree/refactor-elasticity-utils) which also introduces unit tests for all these functions.

**Comment by SiVoelkl** [comment]
We should continue on this to unify formulations for 2D and 3D.

For me it would be important that we keep the formulations in a continuum mechanics sense, meaning that we should set everything up in such a way that we can easily replace building blocks/ functions to include new functionality, e.g. we want to include a new way of computing the strains and we just have to create a function that does this and hopefully everything works out of the box. :wink:

This means that we should have a chain of dependencies like this

1. Predict the displacement $\mathbf{u}$
2. Compute the deformation gradient $\mathbf{F}(\mathbf{u})$
3. Compute strains (e.g. Green-Lagrange) $\mathbf{E}(\mathbf{F})$
4. Compute stresses (according to material law, e.g. St.-Venant) $\mathbf{P}(\mathbf{E})$
5. Compute other stresses in dependency of $\mathbf{P}$

For this to be the same for 2D and 3D we should restructure our inputs and outputs into the network. My idea would be to have them as discrete tensors of shape $[n_{nodes}, dim]$ for the discrete values of a 1. order tensor, $[n_{nodes}, dim, dim]$ for the discrete values of a 2. order tensor, etc.. This would simplify a lot of functions as they would be the same for 2D and 3D.
